drivers/ethernet/phy/phy_marvell_88e1xxx.c

```diff
@@ -0,0 +1,624 @@
+		if (fdx) {
+			state->speed = LINK_FULL_1000BASE_T;
+		} else {
+			state->speed = LINK_HALF_1000BASE_T;
+		}
+	}
+	state->is_up = link_up;
+
+	k_mutex_unlock(&dev_data->mutex);
+
+	if (memcmp(&old_state, state, sizeof(struct phy_link_state)) != 0) {
+		LOG_DBG("%s: PHY link is %s", dev->name, state->is_up ? "up" : "down");
+		if (state->is_up) {
+			LOG_DBG("%s: PHY configured for %s MBit/s %s", dev->name,
+				PHY_LINK_IS_SPEED_1000M(state->speed)  ? "1000"
+				: PHY_LINK_IS_SPEED_100M(state->speed) ? "100"
+								       : "10",
+				PHY_LINK_IS_FULL_DUPLEX(state->speed) ? "FDX" : "HDX");
+		}
+	}
+
+	return ret;
+}
+
+static int phy_marvell_alaska_link_cb_set(const struct device *dev, phy_callback_t cb,
+					  void *user_data)
+{
+	struct marvell_alaska_data *dev_data = dev->data;
+
+	int ret = 0;
+
+	ret = k_mutex_lock(&dev_data->mutex, K_FOREVER);
+	if (ret) {
+		LOG_ERR("%s: set link state callback: mutex lock error", dev->name);
+		return ret;
+	}
+
+	dev_data->cb = cb;
+	dev_data->cb_data = user_data;
+
+	k_mutex_unlock(&dev_data->mutex);
+
+	/* Initial state propagation to the newly registered callback function */
+	phy_marvell_alaska_get_link(dev, &dev_data->state);
+	dev_data->cb(dev, &dev_data->state, dev_data->cb_data);
+
+	return 0;
+}
+
+static void phy_marvell_alaska_monitor_work_handler(struct k_work *work)
+{
+	struct k_work_delayable *dwork = k_work_delayable_from_work(work);
+	struct marvell_alaska_data *dev_data =
+		CONTAINER_OF(dwork, struct marvell_alaska_data, phy_monitor_work);
+	const struct device *dev = dev_data->dev;
+
+	struct phy_link_state state = {};
+	int ret = phy_marvell_alaska_get_link(dev, &state);
+
+	if (ret == 0 && memcmp(&state, &dev_data->state, sizeof(struct phy_link_state)) != 0) {
+		memcpy(&dev_data->state, &state, sizeof(struct phy_link_state));
+		if (dev_data->cb) {
+			dev_data->cb(dev, &dev_data->state, dev_data->cb_data);
+		}
+	}
+
+	k_work_reschedule(&dev_data->phy_monitor_work, K_MSEC(CONFIG_PHY_MONITOR_PERIOD));
+}
+
+static int phy_marvell_alaska_init(const struct device *dev)
+{
+	const struct marvell_alaska_config *const dev_conf = dev->config;
+	struct marvell_alaska_data *dev_data = dev->data;
+
+	int ret;
+
+	dev_data->dev = dev;
+
+	ret = k_mutex_init(&dev_data->mutex);
+	if (ret) {
+		LOG_ERR("%s: init PHY: initialize mutex failed", dev->name);
+		return ret;
+	}
+
+	mdio_bus_enable(dev_conf->mdio_dev);
+
+	k_work_init_delayable(&dev_data->phy_monitor_work, phy_marvell_alaska_monitor_work_handler);
+	k_work_reschedule(&dev_data->phy_monitor_work, K_MSEC(CONFIG_PHY_MONITOR_PERIOD));
+
+	LOG_DBG("%s: init PHY: completed", dev->name);
+
+	return 0;
+}
+
+static DEVICE_API(ethphy, marvell_alaska_phy_api) = {.get_link = phy_marvell_alaska_get_link,
+						     .cfg_link = phy_marvell_alaska_cfg_link,
+						     .link_cb_set = phy_marvell_alaska_link_cb_set,
+						     .read = phy_marvell_alaska_read,
+						     .write = phy_marvell_alaska_write};
```

Comment:
Nit, put a trailing comma after the last entry so it can be formatted like: (same for other driver)

```c
static DEVICE_API(ethphy, marvell_alaska_phy_api) = {
	.get_link = phy_marvell_alaska_get_link,
	.cfg_link = phy_marvell_alaska_cfg_link,
	.link_cb_set = phy_marvell_alaska_link_cb_set,
	.read = phy_marvell_alaska_read,
	.write = phy_marvell_alaska_write,
};
```

Reply:
done, thanks

Reply:
Not fully yet, move the function after `{` to a new line with a single indent.

Reply:
I changed the formatting as per your request, however: if I didn't botch the install of the relevant tooling, the original formatting should have been a direct result of Zephyr's own clang-format file. Maybe there's some changes to be made there?